aptos-move/aptos-vm-types/src/storage/io_pricing.rs

```diff
@@ -166,26 +165,45 @@ impl StoragePricingV2 {
 
 // No storage curve. New gas parameter representation.
 #[derive(Debug, Clone)]
-pub struct StoragePricingV3 {
+pub struct IoPricingV3 {
     pub feature_version: u64,
-    pub free_write_bytes_quota: NumBytes,
+    pub legacy_free_write_bytes_quota: NumBytes,
 }
 
-impl StoragePricingV3 {
+impl IoPricingV3 {
+    fn read_size(&self, loaded: NumBytes) -> NumBytes {
+        if self.feature_version >= 12 {
```

Comment:
you have multiple places with `if self.feature_version >= 12`, what's the logic when you introduce new version (i.e. IoPricingV4) vs having these conditions inside?

Reply:
if I added `v4` it complains that v3 and v4 returns different opaque types here, although they are both `        STORAGE_IO_PER_STATE_SLOT_READ * NumArgs + STORAGE_IO_PER_STATE_BYTE_READ * NumBytes` only differing in the specific numbers of NumArgs and NumBytes.. 
: https://github.com/aptos-labs/aptos-core/blob/b1f3abec3988b0c791fe19b5c9663b75ec1d9393/aptos-move/aptos-vm-types/src/storage/io_pricing.rs#L256 

I got lazy and just updated v3..
cc @vgao1996

Reply:
I guess I can do 
```
   V3(v3) => Either::Right(Either::Left(v3.calculate_read_gas(bytes_loaded))),
   V4(v4) => Either::Right(Either::Right(v4.calculate_read_gas(bytes_loaded)))
```
?

Do you prefer that?

Reply:
since some params are not part of "V4" , I prefer having separate V3 and V4 in this case